src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -1538,13 +1538,22 @@ public RedisResult ScriptEvaluate(LuaScript script, object? parameters = null, C
 
         public RedisResult ScriptEvaluate(LoadedLuaScript script, object? parameters = null, CommandFlags flags = CommandFlags.None)
         {
-            return script.Evaluate(this, parameters, null, flags);
+            return script.Evaluate(this, parameters, withKeyPrefix: null, flags);
         }
 
-        public Task<RedisResult> ScriptEvaluateAsync(string script, RedisKey[]? keys = null, RedisValue[]? values = null, CommandFlags flags = CommandFlags.None)
+        public async Task<RedisResult> ScriptEvaluateAsync(string script, RedisKey[]? keys = null, RedisValue[]? values = null, CommandFlags flags = CommandFlags.None)
         {
             var msg = new ScriptEvalMessage(Database, flags, script, keys, values);
-            return ExecuteAsync(msg, ResultProcessor.ScriptResult, defaultValue: RedisResult.NullSingle);
+
+            try
+            {
+                return await ExecuteAsync(msg, ResultProcessor.ScriptResult, defaultValue: RedisResult.NullSingle);
```

Comment:
I just copied the `ScriptEvaluate`, however not sure if the fallback is even needed here. 

When debugging and restarting the Redis the catch branch was never reached.